### PR TITLE
libreoffice: add templates to dolphin "create new" menu

### PR DESCRIPTION
**Pull request by hqurve**
###### Description of changes

Add templates to dolphin "create new" menu.

Taken from [arch](https://github.com/archlinux/svntogit-packages/blob/6ff67138333a448b03c32a55526750ee7d60d8c8/trunk/PKGBUILD#L355) which takes it from [debian](https://salsa.debian.org/libreoffice-team/libreoffice/libreoffice/-/blob/609ada477ad56e528918c73962c65621e27bb015/rules#L2911) 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hqurve** [comment]
Removed space for consistency

**Comment by 7c6f434c** [comment]
If it is a local convenience addition to LibreOffice, shouldn't it go to the wrapper?

File: pkgs/applications/office/libreoffice/wrapper.nix [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
-  done
-'') // {
-  inherit libreoffice dbus;
-  meta = libreoffice.meta;
+  inherit dbus libreoffice bash;
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
```
Comment (jtojnar) [inline pkgs/applications/office/libreoffice/wrapper.nix:13]: When overriding `installPhase`, you should also provide `runHook preInstall` and `runHook postInstall`. Though, in this case, it is better to rely on the default `installPhase`. See <https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md> for more info.

Comment (hqurve) [reply pkgs/applications/office/libreoffice/wrapper.nix:13]: Thanks. This is a wrapper previously used `runCommand`. However, I thought I had to change it to `mkDerivation` to get access to the unpackPhase to extract the necessary file. But thanks for the link, I didnt know that I could manually unpack the file using `unpack`.

Going to switch it back to runCommand

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -27629,6 +27629,7 @@ with pkgs;
         kdeIntegration = true;
         variant = "fresh";
       });
+    dolphinTemplates = true;
```
Comment (7c6f434c) [inline pkgs/top-level/all-packages.nix:27632]: I think this should now be passed to the wrapper?

Comment (hqurve) [reply pkgs/top-level/all-packages.nix:27632]: ???? I dont understand what youre saying. Could you rephrase it?

Comment (7c6f434c) [reply pkgs/top-level/all-packages.nix:27632]: Oops, I mixed up things. The correct question is whether this should apply to both wrapped versions of LibreOffice

Comment (hqurve) [reply pkgs/top-level/all-packages.nix:27632]: Oh, yeah why not. My initial idea was that people who are using dolphin are most likely use choosing libreoffice-qt over libreoffice. But it won't hurt to have it enabled by default

**Comment by 7c6f434c** [comment]
I never remember if `libreoffice-qt` is built on Hydra (OK, neither I use any DEs…)
